src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -137,7 +151,13 @@ export const NavSearchBar: React.FC<NavSearchBarProps> = ({ venueId }) => {
     if (!searchQuery) return [];
 
     return relatedEvents
-      .filter((event) => event.name.toLowerCase().includes(searchQuery))
+      .filter((event) => {
+        const isRoomEnabled = isDefined(event.room)
+          ? relatedRoomsByTitle[event.room]?.isEnabled ?? true
+          : true;
```

Comment:
current version does not exclude events that are related to disabled venues because:
1) room field can be an empty string and it means that event is attached to a venue (looks like we need to change something in events creating/editing)
2) `relatedRoomsByTitle[event.room]?.isEnabled ?? true` this condition was `true` for the events whose rooms were not added to `relatedRoomsByTitle` ( `elatedRoomsByTitle[event.room]?.isEnabled` === `undefined`)

looks like you have already filtered rooms by `isEnabled` so we can just check here if the room exists in the list.
```suggestion
        const isRoomEnabled =
          isDefined(event.room) && event.room !== ""
            ? isDefined(relatedRoomsByTitle[event.room])
            : true;
```

Reply:
> 2) `relatedRoomsByTitle[event.room]?.isEnabled ?? true` this condition was `true` for the events whose rooms were not added to `relatedRoomsByTitle` ( `relatedRoomsByTitle[event.room]?.isEnabled` === `undefined`)

If I remember correctly, I believe that's an explicit choice I made to make the logic work. The name of the variable isn't the best probably.. but I believe the logic here should be "if there is a room and it is enabled, or if the event is not linked to a room, keep it"

Reply:
Not sure the suggested changes are correct to make.

Reply:
> room field can be an empty string and it means that event is attached to a venue 

So off the top of my head we need to basically handle:

- event without room (eg. linked to venue)
- event with undefined room (this is probably the same thing as above)
- event with room set to `""` (this is probably the same thing as above)
- event with an explicit room, and that room `isEnabled`

---

> (looks like we need to change something in events creating/editing)

This is part of our legacy tech debt patterns of avoiding using `undefined` and similar on backend functions. Definitely something to be changed in future, but well and truly OOS of this issue. It'll be a big/painful refactor I believe.

Reply:
> looks like you have already filtered rooms by `isEnabled` so we can just check here if the room exists in the list.

If that's the case, then the variable name and such may need to change here? I'd need to see the code in context to be 100%.

Reply:
@sunny-viktoryia @0xdevalias fixed event find logic in 3948212 as per Viktoria's suggestion, did really help to hide events that are based in disabled rooms
All the other cases that @0xdevalias described are covered, so that should be fine
User can still point an event to a disabled/non-existing venue via room url, but don't think that we can fix it unless we re do the event create/edit mode as per @sunny-viktoryia comment

Reply:
> User can still point an event to a disabled/non-existing venue via room url, but don't think that we can fix it unless we re do the event create/edit mode as per @sunny-viktoryia comment

That's fine a fine edgecase to leave out for now.